jfrog-cli/artifactory/commands/buildinfo/addartifacts.go

```diff
@@ -0,0 +1,73 @@
+package buildinfo
+
+import (
+	"errors"
+	"github.com/jfrog/jfrog-cli-go/jfrog-cli/artifactory/utils"
+	"github.com/jfrog/jfrog-cli-go/jfrog-cli/artifactory/spec"
+	"github.com/jfrog/jfrog-cli-go/jfrog-cli/utils/config"
+	"github.com/jfrog/jfrog-cli-go/jfrog-client/utils/errorutils"
+	"github.com/jfrog/jfrog-cli-go/jfrog-client/utils/log"
+	"github.com/jfrog/jfrog-cli-go/jfrog-client/artifactory/buildinfo"
+	clientutils "github.com/jfrog/jfrog-cli-go/jfrog-client/artifactory/services/utils"
+)
+
+func AddArtifacts(addArtifactsSpec *spec.SpecFiles, configuration *BuildAddArtifactsConfiguration) (successCount int, failCount int, err error) {
+	log.Info("Running Build Add Artifacts command...")
+	if !configuration.DryRun {
+		buildArtifacts, err := searchArtifacts(configuration.ArtDetails, addArtifactsSpec); if err != nil {
+			return 0, 0, err
+		}
+
+		err = utils.SaveBuildGeneralDetails(configuration.BuildName, configuration.BuildNumber); if err != nil {
+			return 0, len(buildArtifacts), err
+		}
+
+		populateFunc := func(partial *buildinfo.Partial) {
+			partial.Artifacts = buildArtifacts
+		}
+		err = utils.SavePartialBuildInfo(configuration.BuildName, configuration.BuildNumber, populateFunc); if err != nil {
+			return 0, len(buildArtifacts), err
+		}
+
+		return len(buildArtifacts), 0, err
+	} else {
+		return 0, 0, nil
```

Comment:
There's no reason to always return 0, 0 if dry-run is used. The information about the total of affected files is useful also  with dry-run. You look at the *AddDependencies* function (used by the build-add-dependencies command) as a reference. It makes sense for the two commands (build-add-dependencies and build-add-artifacts) to be consistent in terms of behavior.

Reply:
In a dry-run, I shouldn't actually query the Artifactory server, right? Since querying Artifactory is necessary to get a list of affected artifacts (as opposed to the `build-add-dependencies` command that obtains this from the local file system), I don't see how I can return anything else.